crypto/fipsmodule/sha/internal.h

```diff
@@ -19,12 +19,27 @@
 extern "C" {
 #endif
 
-// SHA3_256 constants
+// SHA3 constants
 #define KECCAK1600_WIDTH 1600
+
+#define SHA3_224_CAPACITY_BYTES 56
+#define SHA3_224_DIGEST_BITLENGTH 224
+#define SHA3_224_DIGEST_LENGTH 28
+
 #define SHA3_256_CAPACITY_BYTES 64
 #define SHA3_256_DIGEST_BITLENGTH 256
 #define SHA3_256_DIGEST_LENGTH 32
+
+#define SHA3_384_CAPACITY_BYTES 96
+#define SHA3_384_DIGEST_BITLENGTH 384
+#define SHA3_384_DIGEST_LENGTH 48
+
+#define SHA3_512_CAPACITY_BYTES 128
+#define SHA3_512_DIGEST_BITLENGTH 512
+#define SHA3_512_DIGEST_LENGTH 64
+
 #define SHA3_BLOCKSIZE(bitlen) (KECCAK1600_WIDTH - bitlen * 2) / 8
+#define SHA3_MIN_CAPACITY_BYTES 56
```

Comment:
```suggestion
#define SHA3_MIN_CAPACITY_BYTES SHA3_224_CAPACITY_BYTES
```

you can also statically assert this is the lowest value.

Reply:
Should we explain why this value is used?

Reply:
I can add:
`// buf should be at least of size |rate|` if the previous constant explanation is included. (but I guess that would be too much information inside the code, I added only 
```
// The |SHA3_MIN_CAPACITY_BYTES| corresponds to the lowest security level capacity.
// |buf| should be at least |KECCAK1600_WIDTH /8 - SHA3_MIN_CAPACITY_BYTES| bytes.
```